lib/chewy/search/response.rb

```diff
@@ -33,6 +39,17 @@ def max_score
         @max_score ||= hits_root['max_score']
       end
 
+      # Response `errors` field. Returns `nil` if there is no error.
+      # @return [Array, nil]
+      def errors
+        errors = Array(@body['errors'])
+        # there's another case when failures could be present in some shards, in that case es returns them
+        # in @body[_shards][failures] array
+        errors += Array(@body['_shards']['failures']) if @body['_shards']
```

Comment:
Found another case of failure in ES query not being propagated properly, and handled it as well...

Basically, whenever only some and not all shards experience errors, chewy skips the failures.

```ruby
irb(main):6957> que = Chewy::Search::Request.new(PeopleIndex::Person, ContactsIndex0).query(ids: {values: []}).order("invalid-key")
=> <Chewy::Search::Request {:index=>["contacts_0", "people"], :body=>{:sort=>["invalid-key"], :query=>{:ids=>{:values=>[]}}}}>
irb(main):6958> 
irb(main):6959> 
irb(main):6960> que.hits
D, [2024-10-29T17:36:42.564998 #55546] DEBUG -- Chewy:   [PeopleIndex::Person, ContactsIndex0::Contact] Search (1146.1ms) {:index=>["contacts_0", "people"], :body=>{:sort=>["invalid-key"], :query=>{:ids=>{:values=>[]}}}}
=> []
irb(main):6961> 
irb(main):6962> 
irb(main):6963> que.errors
=> 
[{"shard"=>0,
  "index"=>"contacts_0",
  "node"=>"vdyNxz69SeaF9dU3967T1Q",
  "reason"=>{"type"=>"query_shard_exception", "reason"=>"No mapping found for [invalid-key] in order to sort on", "index_uuid"=>"8i3SbflTR2mV5SWiSNkSAw", "index"=>"contacts_0"}}]
irb(main):6964> 
irb(main):6965> 
irb(main):6969> 
irb(main):6970> Chewy.client.search(que.render)
=> 
{"took"=>4,
 "timed_out"=>false,
 "_shards"=>
  {"total"=>81,
   "successful"=>80,
   "skipped"=>80,
   "failed"=>1,
   "failures"=>
    [{"shard"=>0,
      "index"=>"contacts_0",
      "node"=>"vdyNxz69SeaF9dU3967T1Q",
      "reason"=>{"type"=>"query_shard_exception", "reason"=>"No mapping found for [invalid-key] in order to sort on", "index_uuid"=>"8i3SbflTR2mV5SWiSNkSAw", "index"=>"contacts_0"}}]},
 "hits"=>{"total"=>{"value"=>0, "relation"=>"eq"}, "max_score"=>0.0, "hits"=>[]}}
irb(main):6971> 
irb(main):6972> 
```